centreon/src/Core/Common/Infrastructure/Repository/ApiRepositoryTrait.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * Copyright 2005 - 2023 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types = 1);
+
+namespace Core\Common\Infrastructure\Repository;
+
+trait ApiRepositoryTrait
+{
+    protected ?string $proxy = null;
+
+    protected ?string $url = null;
+
+    protected string $authenticationToken = '';
+
+    protected int $timeout = 60; // Default timeout
+
+    protected int $maxItemsByRequest = 100;
+
+    protected int $maxQueryStringLength = 2048;
+
+    public function setProxy(string $proxy): void
+    {
+        $this->proxy = $proxy;
+    }
+
+    public function setUrl(string $url): void
+    {
+        $this->url = rtrim($url, DIRECTORY_SEPARATOR);
```

Comment:
Url, not path (windos/linux).
```suggestion
        $this->url = rtrim($url, '/');
```